### ISSUE TITLE
Taking a stab a running this on pantheon -- and failing miserably

### ISSUE THREAD

**Issue by turnbuga** [auth0/auth0-drupal#24]
Hello auth0-drupal team,

First thank you for your efforts to integrate auth0 into Drupal - I have it up and integrated successfully on my local system -- using our Auth0 Gold to integrate to a SAML based system, as well as opening up logins for facebook and google.

I run into an error when trying to load the module to our pantheon hosting.

1st - I had to remove the .git and .gitignore as submodules are not supported, so I will maintain the code locally and push updates. Now the module is at least detected and can be installed.

the real issue is that when attempting to login I receive the error:

Fatal error: Class 'OAuth2\Client' not found in /srv/bindings/36b2039eeede484288b73d30a7fdba39/code/sites/all/modules/auth0-drupal/vendor/auth0/auth0-php/src/Auth0.php on line 206

Though I know that adoy's Oauth2 is included in the module.

Any insight or hints on where to look are greatly appreciated.

Cheers,
Gregg

**Comment by glena** [comment]
you need to run composer install to download the module dependencies, I think you need to use something like this https://www.drupal.org/project/composer_manager

**Comment by turnbuga** [comment]
you are absolutely correct @glena - I was using composer from command line on the local system so thought those dependencies were included in the push .. but not so. 

Now just need to figure out how to run composer install on pantheon environment. we could close this out as an issue - though I'll update it when I figure out how to run this remotely.

Thanks indeed.

<img width="1313" alt="screen shot 2016-07-05 at 4 30 37 pm" src="https://cloud.githubusercontent.com/assets/14809023/16602017/26215a92-42ce-11e6-862f-f4a6bc2e60bc.png">

**Comment by glena** [comment]
Did you check the module I said before? That should do the trick

**Comment by glena** [comment]
I think this module should install the dependencies somehow. I am not familiar with it tho

**Comment by turnbuga** [comment]
Found the missing piece. The screen shot mentioned in a previous note in the thread was indeed Composer Manager. Using composer I installed all dependencies for auth0_drupal:
<img width="1130" alt="screen shot 2016-07-06 at 9 35 27 am" src="https://cloud.githubusercontent.com/assets/14809023/16623943/0e98f922-435d-11e6-859b-88c100146f40.png">

- but was still getting the error message. It turns out the all/vendor/adoy/auth2 folder was empty (even though composer manager showed it installed). just copied over the contents of that folder from my local host and now I can use auth0_drupal on pantheon. Thanks again @glena.

**Comment by glena** [comment]
cool :)

**Comment by turnbuga** [comment]
Just a quick update.. had to back out of the idea of using composer_manager as it does not play nice with pantheon - auth0 worked for a while but one of my next commits triggered a rebuild and it wiped out the dependencies I had loaded. Ended up just using composer locally to build, and removed the .git / .gitignore items from the dependencies and included them in the pantheon repo. Pantheon does not currently support submodules and ignores any folders that have a .git included. 

There may be a cleaner way to reload git so it sees the updates but I just did as so:

- removing the local .git and .git ignores in the auth0-drupal/vendor subfolders (dependencies)
- moved the auth0-drupal module folder outside the project
- ran git commit
- moved the module folder 'auth0-drupal' back to sites/all/modules/
- ran git add sites/all/modules/auth0-drupal. 
- All code now showed up in the next git commit, and once pushed to pantheon auth0 worked. 

May help someone.
